src/support/hash.h

```diff
@@ -61,6 +61,24 @@ template<typename T1, typename T2> struct hash<pair<T1, T2>> {
   }
 };
 
+// And hashing tuples is useful, too.
+template<typename T, typename... Ts> struct hash<tuple<T, Ts...>> {
+private:
+  template<size_t... I>
+  static void rehash(const tuple<T, Ts...>& tup,
+                     size_t& digest,
+                     std::index_sequence<I...>) {
+    (wasm::rehash(digest, std::get<1 + I>(tup)), ...);
```

Comment:
Interesting, how does this work? I see this `(CODE, ...)` pattern used in https://en.cppreference.com/w/cpp/utility/integer_sequence but I don't see an explanation for it.

Reply:
This is a [fold expression](https://en.cppreference.com/w/cpp/language/fold) that folds the comma operator over the parameter pack, so it just ends up calling `wasm::rehash` with every value in `I`.

Reply:
TIL, thanks!